### PR TITLE
Add tests for Inputs.Stack in transaction_logic/mina_transaction_logic.ml

### PR DESCRIPTION
**Pull request by gltrost**
Explain your changes:

Inside `transaction_logic/mina_transaction_logic.ml` is a module `Inputs.Stack`. This branch adds tests for `Inputs.Stack` to `transaction_logic/tests/mina_transaction_logic.ml`.

Explain how you tested your changes:

I ran `dune runtest src/lib/transaction_logic` to ensure all tests worked.

Closes [Write tests for Mina_stack module #12593 ](https://github.com/MinaProtocol/mina/issues/12593).

### CODE REVIEW

File: src/lib/transaction_logic/test/transaction_logic.ml [hunk 1]
```diff
@@ -239,3 +239,41 @@ let%test_module "Test transaction logic." =
...
             (run_zkapp_cmd ~fee_payer ~fee ~accounts txns) )
   end )
+
+(* This module tests Inputs.Stack *)
+let%test_module "Test stack module" =
+  ( module struct
+    module Stack = Transaction_logic.For_tests.Stack (Int)
+
+    let%test_unit "Ensure pop works on non-empty list." =
+    Quickcheck.test ~trials
```
Comment (ejMina226) [inline src/lib/transaction_logic/test/transaction_logic.ml:249]: Should this not be indented?

Comment (gltrost) [reply src/lib/transaction_logic/test/transaction_logic.ml:249]: I believe you're correct. The formatting looks off. I'll fix this now.

File: src/lib/transaction_logic/test/transaction_logic.ml [hunk 2]
```diff
@@ -239,3 +239,41 @@ let%test_module "Test transaction logic." =
...
+
+(* This module tests Inputs.Stack *)
+let%test_module "Test stack module" =
+  ( module struct
+    module Stack = Transaction_logic.For_tests.Stack (Int)
+
+    let%test_unit "Ensure pop works on non-empty list." =
+    Quickcheck.test ~trials 
+    (let open Quickcheck in
+    let open Generator.Let_syntax in
```
Comment (ejMina226) [inline src/lib/transaction_logic/test/transaction_logic.ml:251]: Can these two lines not be combined with` let open Quickcheck.Let_syntax `?

**Review by ejMina226 — APPROVED** [review]

File: src/lib/transaction_logic/test/transaction_logic.ml (hunk 1 above)
Comment (gltrost) [reply src/lib/transaction_logic/test/transaction_logic.ml]: Fixed with commit 02b8b867c660c114871de79bda2dc2344e3e7d49.

File: src/lib/transaction_logic/test/transaction_logic.ml (hunk 2 above)
Comment (gltrost) [reply src/lib/transaction_logic/test/transaction_logic.ml]: I'll keep it as for a couple of reasons: 
1. I need to open `Quickcheck` already to use `Generator` functions, so might as well leave the first line as is.
2. In the case that `Generator.let_syntax` diverges from `Quickcheck.let_syntax` in a later version of the library, I'd rather stick with the former in this context.

**Review by mrmr1993 — APPROVED** [review]

File: src/lib/transaction_logic/mina_transaction_logic.ml [hunk 3]
```diff
@@ -546,7 +546,17 @@ module type S = sig
     -> ledger
     -> bool Or_error.t
 
-  module For_tests : sig
+    module For_tests : sig
+      module Stack (Elt : sig type t end) : sig
+        type t = Elt.t list
+        val if_ : bool -> then_:'a -> else_:'a -> 'a
+        val empty : unit -> 'a list
+        val is_empty : 'a list -> bool
```
Comment (rbonichon) [inline src/lib/transaction_logic/mina_transaction_logic.ml]: It probably should read
```
val empty: unit -> t
val is_empty: t -> bool
```

File: src/lib/transaction_logic/mina_transaction_logic.ml [hunk 4]
```diff
@@ -546,7 +546,17 @@ module type S = sig
     -> ledger
     -> bool Or_error.t
 
-  module For_tests : sig
+    module For_tests : sig
+      module Stack (Elt : sig type t end) : sig
+        type t = Elt.t list
+        val if_ : bool -> then_:'a -> else_:'a -> 'a
```
Comment (rbonichon) [inline src/lib/transaction_logic/mina_transaction_logic.ml]: It probably is `if_ : bool -> then_:t -> else_:t -> t` ?

File: src/lib/transaction_logic/mina_transaction_logic.ml [hunk 5]
```diff
@@ -546,7 +546,17 @@ module type S = sig
...
+        val empty : unit -> 'a list
+        val is_empty : 'a list -> bool
+        val pop_exn : t -> Elt.t * t
+        val pop : t -> (Elt.t * t) option
+        val push : Elt.t -> onto:Elt.t list -> t
```
Comment (rbonichon) [inline src/lib/transaction_logic/mina_transaction_logic.ml]: `val push : Elt.t -> onto:t -> t`

**Review by rbonichon — COMMENTED** [review]
Minor changes in exposed type signatures might give more consistency.

File: src/lib/transaction_logic/mina_transaction_logic.ml (hunk 3 above)
Comment (gltrost) [reply src/lib/transaction_logic/mina_transaction_logic.ml]: Good catch, thanks! The way I wrote the signature with polymorphic types was not correct.

File: src/lib/transaction_logic/mina_transaction_logic.ml (hunk 4 above)
Comment (gltrost) [reply src/lib/transaction_logic/mina_transaction_logic.ml]: Fixed with [9103b44](https://github.com/MinaProtocol/mina/pull/12796/commits/9103b44dffae591f1190cc9d457dffc25b7fab88).

File: src/lib/transaction_logic/mina_transaction_logic.ml (hunk 5 above)
Comment (gltrost) [reply src/lib/transaction_logic/mina_transaction_logic.ml]: Fixed with [9103b44](https://github.com/MinaProtocol/mina/pull/12796/commits/9103b44dffae591f1190cc9d457dffc25b7fab88).

**Review by rbonichon — APPROVED** [review]

**Comment by kantp** [comment]
Thanks @robinbb, I had lost track of this one, it seems it was blocked by CI and now got some conflicts. I'll add a task to our backlog to see if we can get it to merge.

**Comment by joaosreis** [comment]
!ci-build-me

**Comment by joaosreis** [comment]
!ci-build-me

**Pull request merged**